docs/source/frameworks/onnx.rst

```diff
@@ -0,0 +1,460 @@
+
+      # Initialize model with the pretrained weights
+      map_location = lambda storage, loc: storage
+      if torch.cuda.is_available():
+	  map_location = None
+      torch_model.load_state_dict(model_zoo.load_url(model_url, map_location=map_location))
+
+      # set the model to inference mode
+      torch_model.eval()
+
+
+   Exporting a model to onnx in PyTorch works via tracing or
+   scripting. In this tutorial we will export a model using
+   tracing. Note how we export the model with an input of
+   ``batch_size=1``, but then specify the first dimension as dynamic
+   in the ``dynamic_axes`` parameter in ``torch.onnx.export()``. The
+   exported model will thus accept inputs of size ``[batch_size, 1,
+   224, 224]`` where ``batch_size`` can vary among each inference.
+
+   .. code-block:: python
+
+      batch_size = 1 # can be any number
+      # Tracing input to the model
+      x = torch.randn(batch_size, 1, 224, 224, requires_grad=True)
+
+      # Export the model
+      torch.onnx.export(torch_model,
+			x,
+			"super_resolution.onnx",   # where to save the model (can be a file or file-like object)
+			export_params=True,        # store the trained parameter weights inside the model file
+			opset_version=10,          # the ONNX version to export the model to
+			do_constant_folding=True,  # whether to execute constant folding for optimization
+			input_names=['input'],   # the model's input names
+			output_names=['output'], # the model's output names
+			dynamic_axes={'input' : {0 : 'batch_size'},    # variable length axes
+				      'output' : {0 : 'batch_size'}})
+
+   Now we can compute the output using ONNX Runtime’s Python APIs:
+
+   .. code-block:: python
+
+      import onnxruntime
+
+      ort_session = onnxruntime.InferenceSession("super_resolution.onnx")
+      # compute ONNX Runtime output prediction
+      ort_inputs = {ort_session.get_inputs()[0].name: to_numpy(x)}
+      # ONNX Runtime will return a list of outputs
+      ort_outs = ort_session.run(None, ort_inputs)
+      print(ort_outs[0])
+
+.. tab:: TensorFlow
+
+   First let's install `tf2onnx <https://github.com/onnx/tensorflow-onnx>`_
+
+   .. code-block:: bash
+
+      pip install tf2onnx
+
+   For this tutorial we will download a pretrained ResNet-50 model:
+
+   .. code-block:: python
+
+      import tensorflow as tf
+      from tensorflow.keras.applications.resnet50 import ResNet50
+
+      model = ResNet50(weights='imagenet')
+
+   Then we can export the model to ONNX format. Notice that we use
+   ``None`` in `TensorSpec
+   <https://www.tensorflow.org/api_docs/python/tf/TensorSpec>`_ to
+   denote the first input dimension as dynamic batch axies, which
+   means this dimension can accept arbitrary input size.
+
+   .. code-block:: python
+
+      spec = (tf.TensorSpec((None, 224, 224, 3), tf.float32, name="input"),)
+      onnx_model, _ = tf2onnx.convert.from_keras(model, input_signature=spec, opset=13)
```

Comment:
you can put this in one line. and run `black` to format the code.

Reply:
But we should do this after 1.0. for docs cleanup and style compliant for docs.
cc @ssheng

Reply:
noted